components/sidebar.jsx

```diff
@@ -685,22 +679,58 @@ export default class Sidebar extends React.Component {
                 {channelLink}
                 {tutorialTip}
             </li>
+
         );
     }
 
-    createChannelButtonOrLink(link, rowClass, icon, displayName, badge, closeButton) {
+    copyURLToClipboard = (e, link) => {
+        e.preventDefault();
+        GlobalActions.showGetChannelLinkModal(link.link);
+    }
+
+    createChannelButtonOrLink(channel, rowClass, icon, displayName, badge, closeButton) {
+        let link = '';
+        if (channel.fake) {
+            link = '/' + this.state.currentTeam.name + '/channels/' + channel.name + '?fakechannel=' + encodeURIComponent(JSON.stringify(channel));
+        } else {
+            link = '/' + this.state.currentTeam.name + '/channels/' + channel.name;
+        }
+
         let element;
         if (isDesktopApp()) {
+            const contextMenu = (
+                <ContextMenu id={'lhs-channel-context-click-menu' + channel.id}>
+                    <MenuItem
+                        data={{link}}
+                        onClick={this.copyURLToClipboard}
+                    >
+                        <FormattedMessage
+                            id='sidebar_right_context.getChannelLink'
+                            defaultMessage='Get Channel Link'
+                        />
+                    </MenuItem>
+                </ContextMenu>
+            );
+
+            const contextMenuTrigger = (
+                <ContextMenuTrigger id={'lhs-channel-context-click-menu' + channel.id}>
+                    <button
+                        className={'btn btn-link ' + rowClass}
+                        onClick={() => this.handleClick(link)}
+                    >
+                        {icon}
+                        <span className='sidebar-item__name'>{displayName}</span>
+                        {badge}
+                        {closeButton}
+                    </button>
+                </ContextMenuTrigger>
+            );
+
             element = (
-                <button
-                    className={'btn btn-link ' + rowClass}
-                    onClick={() => this.handleClick(link)}
-                >
-                    {icon}
-                    <span className='sidebar-item__name'>{displayName}</span>
-                    {badge}
-                    {closeButton}
-                </button>
+                <span>
+                    {contextMenu}
+                    {contextMenuTrigger}
+                </span>
```

Comment:
I suspect that the addition of this `<span />` is what's causing the CSS dissimilarities in the LHS between the webapp UI and the desktop UI. I'm trying to track down the CSS selector that is getting confused by this.

Reply:
@MusikPolice We can wait for Asaad to return next week and have him take a look. 

In addition to styling, can we also match the functionality of the webapp? I.e. 
 - name = `Copy Link Address`
 - clicking the option copies the link to the clipboard

![image](https://user-images.githubusercontent.com/13119842/34579428-f649d824-f156-11e7-8583-6ed8f4cdd938.png)

Reply:
I can change the text in the popover to be "Copy Link Address" to match the text in browser context menus. I do think, however, that we should continue to pop the modal to copy the link out of, as that matches what we do for other links, such as the team invite link.

Reply:
@MusikPolice Why deviate from webapp experience?

Reply:
@jasonblais removed the modal in https://github.com/mattermost/mattermost-webapp/pull/547/commits/a1426ab74800e6fcc0144606fe4023f82c58392f